test/MEVCapturingHook.t.sol

```diff
@@ -38,17 +44,90 @@ contract MEVCapturingHookTest is Test, Deployers {
         // Deploy two test tokens
         (token0, token1) = deployMintAndApprove2Currencies();
 
+        alice = vm.createWallet("alice");
+        bob = vm.createWallet("bob");
+
         // Deploy our hook
-        // TODO: set correct flags
-        uint160 flags = uint160(0);
+        uint160 flags =
+            uint160(Hooks.BEFORE_SWAP_FLAG | Hooks.BEFORE_INITIALIZE_FLAG | Hooks.BEFORE_SWAP_RETURNS_DELTA_FLAG);
         address hookAddress = address(flags);
 
-        deployCodeTo("MEVCapturingHook.sol", abi.encode(manager), hookAddress);
+        deployCodeTo("MEVCapturingHook.sol", abi.encode(manager, address(this)), hookAddress);
         hook = MEVCapturingHook(hookAddress);
+
+        // Init Pool
+        (key,) = initPool(token0, token1, hook, LPFeeLibrary.DYNAMIC_FEE_FLAG, SQRT_PRICE_1_1);
```

Comment:
I think that the DYNAMIC _FEE_FLAG should not be used. I think its wise to always charge a standard fee for every trade, plus for high priority txs the fee donation plus the normal fee.

Right now, you have:

        if (priorityFee < priorityThreshold || block.number == lastTradedBlock[poolId]) {
            return (BaseHook.beforeSwap.selector, BeforeSwapDeltaLibrary.ZERO_DELTA, 0);
        }
        and this would mean that retail traders are charged zero fee, which is not good. 
        You could also stick the dynamic fee flag, but then return a positive fee here.